core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -216,8 +216,8 @@ public void testFromXContent() throws IOException {
 
         QueryBuilder newQuery = queryParserService.queryParser(testQuery.queryId()).fromXContent(context);
         assertNotSame(newQuery, testQuery);
-        assertEquals(newQuery, testQuery);
-        assertEquals(newQuery.hashCode(), testQuery.hashCode());
+        assertEquals("Queries should be equal: " + newQuery + " vs. " + testQuery, newQuery, testQuery);
+        assertEquals("Queries should have equal hashcodes: " + newQuery + " vs. " + testQuery, newQuery.hashCode(), testQuery.hashCode());
```

Comment:
In this particular case I was burnt: Had to open the debugger one time too often to see if it's again those bloody locale settings that blow the hashcode and equals implementation.